lib/util/splice.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Plain old regular Array#splice may cause stackoverflow if we have too many items
+ */
+function chunkedSplice(array, start, deleteCount, items) {
+  var chunkSize = 10000
+  var splice = [].splice
+
+  // Make deleteCount >= 0
+  deleteCount = Math.max(deleteCount, 0)
+  // Make start between 0 and array.length (included)
+  start = Math.min(start, array.length)
+  if (start < 0) {
+    start = Math.max(array.length + start, 0)
+  }
+
+  if (items.length < chunkSize) {
+    // No need to chunk the items in this case
+    return splice.apply(array, [start, deleteCount].concat(items))
```

Comment:
Will mutating help perf here?

```suggestion
    items.unshift(start, deleteCount)
    return splice.apply(array, items)
```

Reply:
Might be bad to mutate, but as this is such an internal function which is called a bazillion times, it might make sense to do funky things like that if it helps

Reply:
mutation might be tricky since we may introduce bugs that are hard to debug. Example from the current commit:

```js
chunkedSplice(events, index, 2, subevents)
index += subevents.length - 1
```